python/dgl/distributed/graph_partition_book.py

```diff
@@ -923,6 +981,38 @@ def etypes(self):
         """
         return self._etypes
 
+    @property
+    def canonical_etypes(self):
+        """Get the list of canonical edge types
+        """
+        return self._canonical_etypes
+
+    def to_canonical_etype(self, etype):
+        """Convert an edge type to the corresponding canonical edge type.
+        If canonical etype is not available, no conversion is applied.
+        """
+        if etype is None:
+            if len(self.etypes) != 1:
+                raise DGLError('Edge type name must be specified if there are more than one '
+                               'edge types.')
+            etype = self.etypes[0]
+        if isinstance(etype, tuple):
+            if etype not in self.canonical_etypes:
+                raise DGLError('Edge type "{}" does not exist.'.format(etype))
+            return etype
+        if not self._etype2canonical:
+            # canonical etype is not available, no conversion is applied.
+            # This is the case that 'etypes' passed in when instantiating
+            # are in format of str instead of (str, str, str).
+            return etype
```

Comment:
we should disable this in the future. can you add TODO to fix it in the future.

Reply:
yes. Totally disable it may take time as such deprecation requires users always use the new format of partition config which is always dumped with canonical etypes.

Reply:
Since this is a new API, can we simply raise error here?